engine.py

```diff
@@ -611,6 +611,43 @@ def _set_project(self):
         self.logger.info("Setting Maya project to '%s'", proj_path)
         pm.mel.setProject(proj_path)
 
+    def get_session_path(self):
+        """
+        Returns the path to the current scene file if it resides on
+        disk. If unsaved, it returns an empty string.
+
+        :returns: Path to the current scene
+        """
+        path = cmds.file(query=True, sceneName=True)
+
+        if isinstance(path, unicode):
+            path = path.encode("utf-8")
+
+        return path
+
+    def get_session_dependencies(self):
+        """
+        Returns a list of file dependencies for this scene.
+
+        :returns: List of files or folder paths constituting the scene dependencies.
+        """
+        dependencies = []
+
+        # include file paths from file references
+        for ref_node in cmds.ls(type="reference"):
```

Comment:
Again, I have the same concerns here that I had for the Nuke engine about filtering or adding missing files.